File: core/src/main/java/com/memoizlabs/kotlin_playground/currency/GBPExchangeRate.java
package com.memoizlabs.kotlin_playground.currency;

public class GBPExchangeRate {

    private final Money toUSD;
    private final Money toEUR;

    public GBPExchangeRate(Money toUSD, Money toEur) {
        this.toUSD = toUSD;
        this.toEUR = toEur;
    }

    public Money getToEUR() {
        return toEUR;
    }

    public Money getToUSD() {
        return toUSD;
    }

    @Override
    public String toString() {
        return "GBPExchangeRate{" +
                "toUSD=" + toUSD +
                ", toEUR=" + toEUR +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GBPExchangeRate)) {
            return false;
        }

        GBPExchangeRate that = (GBPExchangeRate) o;

        if (getToUSD() != null ? !getToUSD().equals(that.getToUSD()) : that.getToUSD() != null) {
            return false;
        }
        return getToEUR() != null ? getToEUR().equals(that.getToEUR()) : that.getToEUR() == null;
    }

    @Override
    public int hashCode() {
        int result = getToUSD() != null ? getToUSD().hashCode() : 0;
        result = 31 * result + (getToEUR() != null ? getToEUR().hashCode() : 0);
        return result;
    }
}

File: core/src/main/java/com/memoizlabs/kotlin_playground/currency/GBPExchangeRate.kt
package com.memoizlabs.kotlin_playground.currency

class GBPExchangeRate(val toUSD: Money, val toEUR: Money) {

    override fun toString(): String {
        return "GBPExchangeRate{" +
                "toUSD=" + toUSD +
                ", toEUR=" + toEUR +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is GBPExchangeRate) {
            return false
        }

        if (if (toUSD != null) toUSD != o.toUSD else o.toUSD != null) {
            return false
        }
        return if (toEUR != null) toEUR == o.toEUR else o.toEUR == null
    }

    override fun hashCode(): Int {
        var result = if (toUSD != null) toUSD.hashCode() else 0
        result = 31 * result + if (toEUR != null) toEUR.hashCode() else 0
        return result
    }
}
